### PR TITLE
Test for cost/stats calculation done by CoefficientBasedStatsCalculator (v2)

### PR DESCRIPTION
**Pull request by kokosing**

### CODE REVIEW

**Comment by kokosing** [comment]
Supersedes #9749

File: presto-tests/src/test/java/com/facebook/presto/tests/TestTpchDistributedStats.java [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+        statisticsAssertion.check("SELECT * FROM lineitem WHERE l_shipdate <= DATE '1998-12-01' - INTERVAL '90' DAY")
+                //TODO fix estimates
+                .estimate(OUTPUT_ROW_COUNT, relativeError(-.55, -.45));
+    }
+
+    @Test
+    public void testJoin()
+    {
+        statisticsAssertion.check("SELECT * FROM  part, partsupp WHERE p_partkey = ps_partkey")
+                .estimate(OUTPUT_ROW_COUNT, relativeError(.95, 1.05));
```
Comment (findepi) [inline presto-tests/src/test/java/com/facebook/presto/tests/TestTpchDistributedStats.java]: At first it looks a bit like ±5%, but it seems to actually mean "expect estimate twice as much as actual value" (i.e. poor estimate). Am i reading this correctly? If so, mark it with `// TODO improve estimate` or something.

File: presto-tests/src/test/java/com/facebook/presto/tests/TestTpchDistributedStats.java [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+    public void testSetOperations()
+    {
+        statisticsAssertion.check("SELECT * FROM nation UNION SELECT * FROM nation")
+                .noEstimate(OUTPUT_ROW_COUNT);
+
+        statisticsAssertion.check("SELECT * FROM nation INTERSECT SELECT * FROM nation")
+                .noEstimate(OUTPUT_ROW_COUNT);
+
+        statisticsAssertion.check("SELECT * FROM nation EXCEPT SELECT * FROM nation")
+                .noEstimate(OUTPUT_ROW_COUNT);
```
Comment (findepi) [inline presto-tests/src/test/java/com/facebook/presto/tests/TestTpchDistributedStats.java]: Run assertion and check there isn't any estimate? Why do we have it in that form?

File: presto-tests/src/main/java/com/facebook/presto/tests/statistics/MetricComparator.java [hunk 3]
```diff
@@ -33,15 +33,16 @@
...
 import static com.google.common.collect.Maps.transformValues;
-import static java.util.Arrays.asList;
 
-public class MetricComparator
+public final class MetricComparator
 {
-    private final List<Metric> metrics = asList(Metric.values());
+    private MetricComparator()
+    {
+    }
```
Comment (findepi) [inline presto-tests/src/main/java/com/facebook/presto/tests/statistics/MetricComparator.java]: `private MetricComparator() {}`

File: presto-tests/src/test/java/com/facebook/presto/tests/TestTpchDistributedStats.java (hunk 2 above)
Comment (kokosing) [reply presto-tests/src/test/java/com/facebook/presto/tests/TestTpchDistributedStats.java]: No support for `EXCEPT` in stats estimation math I think.

**Pull request merged**